tests/ui/escape_analysis.rs

```diff
@@ -154,6 +154,15 @@ impl<T> MyTrait for Box<T> {
     fn do_sth(self) {}
 }
 
+/// Fix for #4804
+///
+/// This shouldn't warn for `boxed_local` as traits don't have a know size in compile time
+trait DefaultTraitImplTest {
+    fn default_impl(self: Box<Self>) -> u32 {
+        5
+    }
+}
```

Comment:
Is this enough to cover this fix? There might be some more niche cases which I hadn't encountered but which should be tested as well.